tcmur_device.c

```diff
@@ -299,7 +299,8 @@ int tcmu_acquire_dev_lock(struct tcmu_device *dev)
 			retries++;
 			goto retry;
 		}
-
+		/* fall through */
+	case TCMUR_LOCK_BLACKLISTED:
 		tcmu_dev_dbg(dev, "Fail handler device connection.\n");
 		tcmu_notify_conn_lost(dev);
```

Comment:
Thanks for the patch and debugging this.

We used to do something similar. The problem was that for failback or failover with N > 2 nodes, this would cause the initiator to failover extra times because the conn lost call kills the session the initiator is trying to use causing it to try another path then later switch back over when the tpg is enabled. I think that is definitely better than data corruption though.

Let me do some more testing in the morning.